contracts/Peeranha.sol

```diff
@@ -49,9 +50,18 @@ contract Peeranha is IPeeranha, Initializable {
         SecurityLib.setupRole(userContext, SecurityLib.DEFAULT_ADMIN_ROLE, msg.sender);
     }
 
-    function getRatingToReward(address user, uint16 rewardPeriod) external view override returns(int32) {
-        RewardLib.PeriodRating storage userPeriod = RewardLib.getUserPeriodRating(userContext.userRewards, user, rewardPeriod);
-        return userPeriod.ratingToReward;
+    function getActiveInCommunity(address user, uint16 rewardPeriod) external override returns(uint32[] memory) {
+        require(
+            !userContext.userRatingCollection.communityRatingForUser[user].userRewards[rewardPeriod].isPaid,
+            "You already picked up this reward."
+        );
+        require(
+            userContext.userRatingCollection.communityRatingForUser[user].userRewards[rewardPeriod].activeInCommunity.length > 0,
+            "No reward for you in this period"
+        );
+
+        userContext.userRatingCollection.communityRatingForUser[user].userRewards[rewardPeriod].isPaid = true;
```

Comment:
This function has to be view only and can't change any values.

In general, it is very confusing when action with name that starts with `get` changes any state.